Aryan9101/app/views.py

```diff
@@ -0,0 +1,31 @@
+from flask import render_template, flash, request, redirect, url_for
+from wtforms import Form, StringField, validators
+from flask import Flask
+from app import app
+
+WTF_CSRF_ENABLED = True #Important Config that prevents CSRF attacks
+
+class GreetingForm(Form):
+    """
+    Create a Greeting form class.
+    Contains the main attributes i.e, String fields to be filled in a form
+    """
+    username = StringField('Username', [validators.Length(min = 1)]) #validator to make sure data is entered
+
+@app.route('/')
```

Comment:
Try changing this to the following:
```
@app.route('/', methods=['POST', 'GET'])
```

Reply:
That was there the entire time. I didn't update my PR yet, but in my personal repo, it is there

Reply:
So the root route has both GET and POST on the heroku version?

Reply:
yup

Reply:
https://github.com/Aryan9101/GCIWebForm

Reply:
@Aryan9101 can you please post your heroku logs here

Reply:
The new ones after making the previous changes

Reply:
The logs when the method not allowed error is thrown

Reply:
![capture2](https://user-images.githubusercontent.com/25390298/33523553-e39c02e2-d7cf-11e7-8ad5-68ebd17825c6.PNG)

Reply:
Are these the logs after encountering the error?

Reply:
@Aryan9101 ^

Reply:
These are the logs after I changed my run.py. I wasn't getting the OSError anymore, but the application still is not working when deployed.

Reply:
@Aryan9101 please try accessing the greeting endpoint or simply refresh your landing page and see whats there in the log